core/src/main/java/com/linecorp/armeria/server/ServerListenerBuilder.java

```diff
@@ -262,6 +265,62 @@ public ServerListenerBuilder whenStopped(Iterable<? extends Consumer<? super Ser
         return this;
     }
 
+    /**
+     * Add a callback that gracefully shuts down the given {@link ExecutorService} when the {@link Server}
+     * is stopping. It will wait indefinitely for the {@link ExecutorService} to terminate during shutdown.
+     */
+    public ServerListenerBuilder shutdownWhenStopping(ExecutorService executorService) {
+        requireNonNull(executorService, "executorService");
+        serverStoppingCallbacks.add(s -> ShutdownSupport.of(executorService).shutdown());
+        return this;
+    }
+
+    /**
+     * Add a callback that gracefully shuts down the given {@link ExecutorService} when the {@link Server}
+     * is stopping. It allows a maximum duration of {@code terminationTimeout} for the {@link ExecutorService}
+     * to terminate gracefully before it is forcefully terminated.
+     */
+    public ServerListenerBuilder shutdownWhenStopping(ExecutorService executorService,
+                                                      Duration terminationTimeout) {
+        requireNonNull(executorService, "executorService");
+        requireNonNull(terminationTimeout, "terminationTimeout");
+        long nanoSeconds;
+        try {
+            nanoSeconds = terminationTimeout.toNanos();
+        } catch (ArithmeticException ignore) {
+            nanoSeconds = terminationTimeout.isNegative() ? Long.MIN_VALUE : Long.MAX_VALUE;
+        }
+        shutdownWhenStopping(executorService, nanoSeconds, TimeUnit.NANOSECONDS);
+        return this;
+    }
+
+    /**
+     * Add a callback that gracefully shuts down the given {@link ExecutorService} when the {@link Server}
+     * is stopping. It allows a maximum duration of {@code terminationTimeout} in the specified
+     * {@code timeUnit} for the {@link ExecutorService} to terminate gracefully before it is forcefully
+     * terminated.
+     */
+    public ServerListenerBuilder shutdownWhenStopping(ExecutorService executorService, long terminationTimeout,
+                                                      TimeUnit timeUnit) {
+        requireNonNull(executorService, "executorService");
+        requireNonNull(timeUnit, "timeUnit");
+        serverStoppingCallbacks.add(
+                s -> ShutdownSupport.of(executorService, terminationTimeout, timeUnit).shutdown());
+        return this;
+    }
+
+    /**
+     * Add a callback that gracefully shuts down the given {@link ExecutorService} when the {@link Server}
+     * is stopping. It uses the specified {@code shutdownStrategy} to control the shutdown behavior.
+     */
+    public ServerListenerBuilder shutdownWhenStopping(ExecutorService executorService,
+                                                      Consumer<ExecutorService> shutdownStrategy) {
```

Comment:
What do you think of removing this API? I don't see that this method is useful as a public API. If customized logic is required, users could easily use `whenStopping(server -> ...; executorService.shutdown())`.